src/window/basic_frame.rs

```diff
@@ -233,24 +230,31 @@ impl Frame for BasicFrame {
         shm: &Proxy<wl_shm::WlShm>,
         implementation: Box<Implementation<u32, FrameRequest> + Send>,
     ) -> Result<BasicFrame, ::std::io::Error> {
-        let pools = DoubleMemPool::new(&shm)?;
         let parts = [
             Part::new(base_surface, compositor, subcompositor),
             Part::new(base_surface, compositor, subcompositor),
             Part::new(base_surface, compositor, subcompositor),
             Part::new(base_surface, compositor, subcompositor),
             Part::new(base_surface, compositor, subcompositor),
         ];
+        let inner = Arc::new(Inner {
+            parts: parts,
+            size: Mutex::new((1, 1)),
+            resizable: Mutex::new(true),
+            implem: Mutex::new(implementation),
+            maximized: Mutex::new(false),
+        });
+        let my_inner = inner.clone();
+        let pools = DoubleMemPool::new(&shm, move |_, _| {
+            my_inner
+                .implem
+                .lock()
+                .unwrap()
+                .receive(FrameRequest::Refresh, 0);
```

Comment:
Please add a comment explaining that we can just blindly send a `Refresh` because the `DoubleMemPool` callback only fires if `DoubleMemPool.pool()` had previously returned `None`, which means a draw was postponed.